src/mds/MDSRank.cc

```diff
@@ -1486,6 +1488,10 @@ void MDSRank::forward_message_mds(const cref_t<MClientRequest>& m, mds_rank_t md
 
   // tell the client where it should go
   auto session = get_session(m);
+  if (!session) {
+    dout(1) << "session was removed, failed to forward message " << mdr << dendl;
```

Comment:
the line looks incomplete; "session was removed" raises concerns like how, where, why. Maybe a better line would be 
```
dout(1) << "no session found, failed to forward message for metadata request " << mdr << dendl;

```